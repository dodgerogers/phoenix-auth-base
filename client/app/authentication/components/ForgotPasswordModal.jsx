import React from 'react'
import PropTypes from 'prop-types';
import AuthenticationModal from './AuthenticationModal';
import ForgotPasswordFormContainer from '../containers/ForgotPasswordFormContainer';
import AuthenticationLinks, { links } from './AuthenticationLinks';


const ForgotPasswordModal = props => {
  return (
    <AuthenticationModal title="Forgot your password" {...props}>
      <ForgotPasswordFormContainer {...props} />
      <AuthenticationLinks exclude={[links.FORGOT_PASSWORD]} />
    </AuthenticationModal>
  );
}

export default ForgotPasswordModal;
