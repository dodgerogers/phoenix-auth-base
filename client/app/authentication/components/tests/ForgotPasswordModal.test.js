import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ForgotPasswordModal from '../ForgotPasswordModal';


describe('ForgotPasswordModal', () => {
  const fullProps = {
    close: jest.fn(),
  };

  const pureComponent = (props) => shallow(
    <ForgotPasswordModal {...props} />
  );

  it('matches snapshot', () => {
    const wrapper = pureComponent(fullProps);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
