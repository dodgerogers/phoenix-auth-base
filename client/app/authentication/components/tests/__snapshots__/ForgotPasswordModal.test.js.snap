// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordModal matches snapshot 1`] = `
<AuthenticationModal
  close={[Function]}
  title="Forgot your password"
>
  <Connect(ReduxForm)
    close={[Function]}
  />
  <AuthenticationLinks
    exclude={
      Array [
        "FORGOT_PASSWORD",
      ]
    }
  />
</AuthenticationModal>
`;
